wled00/const.h

```diff
@@ -44,8 +44,8 @@
       #define WLED_MIN_VIRTUAL_BUSSES 4
     #else
       #if defined(USERMOD_AUDIOREACTIVE)      // requested by @softhack007 https://github.com/blazoncek/WLED/issues/33
-        #define WLED_MAX_BUSSES 8
-        #define WLED_MIN_VIRTUAL_BUSSES 2
+        #define WLED_MAX_BUSSES 9
```

Comment:
This `#ifdef` may be unnecessary as well. Just allow 10 digital buses.

A check in Audioreactive's `setup()` could be added as follows:
```c++
int digitalBuses = 0;
for (int i=0; i<busses.getNumbuses(); i++) if (busses[i]->getType() >= 16 && busses[i]->getType() <= 31) digitalBuses++;
if (digitalBuses>9) return;
```

Reply:
Thanks, that check is actually an easy way to verify. I would still need a new #ifdef in audioreactive, because a similar check for bus 5 is needed for -S2.

I'm still not sure it's a good solution from user experience point of view. Because AR would simply fail to initialize, without information what is wrong. Maybe I'll need to add a flag and put something into info page, but space there is limited so we needed a short but absolutely clear message text.

Need to think about it ...

Reply:
There is no problem if usermod requires core dependency, but core should not be dependant on usermods. That's how I see it.
Info dialog can be used with a simple note: "AR Inactive: Too many LED outputs used."

And perhaps instead of returning in `setup()` set `enabled` to `false`.

Reply:
No, as you do not want SPI variants included.
const.h defines bus types and anything 16>=x<=31 is single pin digital.
This may change of course in the future, but such magic numbers are also used in JS for UI so there is no way around them.

Reply:
Even if a magic number needs to be defined in both C and JavaScript, they should still be constants not magic numbers

Reply:
The *magic* numbers are defined in const.h. Look at LED types `TYPE_xxxxx`.
JS does not have such definitions. Yet.